### ISSUE TITLE
Cannot restore backup: either full or partial fail

### ISSUE THREAD

**Issue by ocalvo** [home-assistant/supervisor#3512]
### Describe the issue you are experiencing

When trying a to restore either a full or partial restore fails.
For example, trying to restore the ssh addon fails with:
```
22-03-12 19:22:25 ERROR (MainThread) [supervisor.jobs] Unhandled exception: 'core_ssh'
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 106, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/addons/__init__.py", line 367, in restore
    await addon.restore(tar_file)
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 832, in restore
    restore_image = self._image(data[ATTR_SYSTEM])
  File "/usr/src/supervisor/supervisor/addons/model.py", line 629, in _image
    return config[ATTR_IMAGE].format(arch=self.arch)
  File "/usr/src/supervisor/supervisor/addons/model.py", line 509, in arch
    if ATTR_IMAGE in self.data:
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 157, in data
    return self.sys_addons.data.system[self.slug]
KeyError: 'core_ssh'
```

### What is the used version of the Supervisor?

supervisor-2022.03.3

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

7.4

### What version of Home Assistant Core is installed?

core-2022.3.0

### Steps to reproduce the issue

(I can provide access to the full back tar file)
1. Do a partial restore of the SSH addon
Expected:
Restore is completed with errors
Actual:
Fails

### Anything in the Supervisor logs that might be useful for us?

```txt
22-03-12 19:22:25 ERROR (MainThread) [supervisor.jobs] Unhandled exception: 'core_ssh'
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 106, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/addons/__init__.py", line 367, in restore
    await addon.restore(tar_file)
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 832, in restore
    restore_image = self._image(data[ATTR_SYSTEM])
  File "/usr/src/supervisor/supervisor/addons/model.py", line 629, in _image
    return config[ATTR_IMAGE].format(arch=self.arch)
  File "/usr/src/supervisor/supervisor/addons/model.py", line 509, in arch
    if ATTR_IMAGE in self.data:
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 157, in data
    return self.sys_addons.data.system[self.slug]
KeyError: 'core_ssh'
```
```


### Additional information

I can provide access to the full backup or some parts of it.

**Comment by dominicairportindoorparking** [comment]
same problem, password protected backup. supervisor 22.03.3

22-03-12 13:49:45 ERROR (MainThread) [supervisor.backups.backup] Can't validate data for /data/tmp/tmpsyfv879p/backup.tar: expected boolean for dictionary value @ data['protected']. Got '8'

**Comment by sklein972** [comment]
I have the same issue with 22.03.4

**Comment by whistlerofficial** [comment]
i also got the same issue with 2022.03.3 (Supervisior and Core)

**Comment by sermayoral** [comment]
I think this is solved with supervisor-2022.03.4. There is a release but it doesn't appear in HA 😢

**Comment by whistlerofficial** [comment]
> I think this is solved with supervisor-2022.03.4. There is a release but it doesn't appear in HA 😢
> 
> Maybe the problem is that https://version.home-assistant.io/stable.json is not updated!!

Should be fixed thanks to https://github.com/home-assistant/version/commit/0e63e44b2c7b139d92c3e6df79396705370b07d1

Just tested to restore and it did work :)
